pkg/hubble/metrics/cell/certloader.go

```diff
@@ -0,0 +1,52 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Hubble
+
+package metricscell
+
+import (
+	"github.com/cilium/hive/cell"
+	"github.com/spf13/pflag"
+
+	"github.com/cilium/cilium/pkg/crypto/certloader"
+	"github.com/cilium/cilium/pkg/promise"
+)
+
+type tlsConfigPromise promise.Promise[*certloader.WatchedServerConfig]
+
+// certloaderGroup provides a promise that can be used to obtain a TLS config
+// capable of automatically sourcing/reloading certificates from disk.
+//
+// We wrap the promise in our own type to avoid conflicts/replacements with other
+// certloader promises. We use a group instead of a module to be able to use
+// cell.ProvidePrivate and avoid providing the promise to the rest of the hive.
+var certloaderGroup = cell.Group(
+	cell.ProvidePrivate(func(cfg certloaderConfig) certloader.Config {
+		return certloader.Config(cfg)
+	}),
+	cell.ProvidePrivate(certloader.NewWatchedServerConfigPromise),
+	cell.ProvidePrivate(func(p promise.Promise[*certloader.WatchedServerConfig]) tlsConfigPromise {
+		return p
+	}),
```

Comment:
Nit/Personal preference(feel free to ignore): I'm not a huge fan of `Provide` chains that pass along intermediate values, it always takes me a second to parse. I think you could also change this to the following, which at least to me is a bit more readable and we completely avoid the problem of providing any certloader package types to the dependency graph:

```go
cell.ProvidePrivate(func(lc cell.Lifecycle, log *slog.Logger, cfg certloaderConfig) (tlsConfigPromise, error) {
	 return certloader.NewWatchedServerConfigPromise(lc, log, certloader.Config(cfg))
}),
```

Reply:
I think this is reasonable and it does make it more succinct, will update.